libs/libc/sched/task_startup.c

```diff
@@ -94,6 +94,11 @@ extern uintptr_t _etext;
 static void cxx_initialize(void)
 {
 #ifdef CONFIG_HAVE_CXXINITIALIZE
+#if defined(CONFIG_ARCH_SIM) && defined(CONFIG_HOST_MACOS)
+  extern void macho_call_saved_init_funcs(void);
```

Comment:
can we move this special hack to sim's up_initialize or board_initialize?

Reply:
i can't think of a better place. neither up_initialize or board_initialize doesn't seem appropriate to me.

i can introduce ARCH_HAVE_CXXINITIALIZE or something along the line if you prefer.

Reply:
> @yamt see my reply. Another approach is switch to our own entry point then all c++ initialization bypass automatically.

IIRC, the program entrypoint is invoked after ctors are executed on macOS.